st2common/st2common/persistence/base.py

```diff
@@ -112,12 +112,52 @@ def aggregate(cls, *args, **kwargs):
         return cls._get_impl().aggregate(*args, **kwargs)
 
     @classmethod
-    def add_or_update(cls, model_object, publish=True, dispatch_trigger=True):
+    def insert(cls, model_object, publish=True, dispatch_trigger=True,
+               log_not_unique_error_as_debug=False):
+        if model_object.id:
+            raise ValueError('id for object %s was unexpected.' % model_object)
+        try:
+            model_object = cls._get_impl().insert(model_object)
+        except NotUniqueError as e:
+            if log_not_unique_error_as_debug:
+                LOG.debug('Conflict while trying to save in DB.', exc_info=True)
```

Comment:
Imo, a better approach would be to not log anything here and only log a a message in the outer most layer. The reason for that is `NotUniqueError` representing a fatal error or not depends on the consumer (code which called this function) and the code which calls it should also be responsible for logging it appropriately.

We already discussed and bike shed this a lot yesterday, but imo, we are not solving the right problem. Solving a right problem in this case would be deciding on consistent conventions for logging exception since the same "problem" exists through the code base.